src/main/java/org/opensearch/knn/index/KNNMethod.java

```diff
@@ -72,13 +72,24 @@ public boolean containsSpace(SpaceType space) {
      * Validate that the configured KNNMethodContext is valid for this method
      *
      * @param knnMethodContext to be validated
+     * @return ValidationException produced by validation errors; null if no validations errors.
      */
-    public void validate(KNNMethodContext knnMethodContext) {
+    public ValidationException validate(KNNMethodContext knnMethodContext) {
+        ValidationException validationException = null;
         if (!containsSpace(knnMethodContext.getSpaceType())) {
-            throw new ValidationException();
+            validationException = new ValidationException();
+            validationException.addValidationError("\"" + this.methodComponent.getName() +
+                    "\" configuration does not support space type: \"" + knnMethodContext.getSpaceType() + "\".");
```

Comment:
Shall we use String.format to build the string?

Reply:
Sure